stackit/internal/services/iaas/publicipassociate/resource.go

```diff
@@ -0,0 +1,360 @@
+			"Both resources have control of the network_interface association. If used together, this will lead to conflicts.",
+	}
+	resp.Schema = schema.Schema{
+		MarkdownDescription: features.AddBetaDescription(fmt.Sprintf("%s\n%s", descriptions["main"], descriptions["warning_message"])),
+		Description:         fmt.Sprintf("%s\n%s", descriptions["main"], descriptions["warning_message"]),
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "Terraform's internal resource ID. It is structured as \"`project_id`,`public_ip_id`,`network_interface_id`\".",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"project_id": schema.StringAttribute{
+				Description: "STACKIT project ID to which the public IP is associated.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"public_ip_id": schema.StringAttribute{
+				Description: "The public IP ID.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+			"ip": schema.StringAttribute{
+				Description: "The IP address.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					validate.IP(),
+				},
+			},
+			"network_interface_id": schema.StringAttribute{
+				Description: "Associates the public IP with a network interface or a virtual IP (ID).",
+				Required:    true,
+				Validators: []validator.String{
+					validate.UUID(),
+					validate.NoSeparator(),
+				},
+			},
+		},
+	}
+}
+
+// Create creates the resource and sets the initial Terraform state.
+func (r *publicIpAssociateResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) { // nolint:gocritic // function signature required by Terraform
+	// Retrieve values from plan
+	var model Model
+	diags := req.Plan.Get(ctx, &model)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	projectId := model.ProjectId.ValueString()
+	publicIpId := model.PublicIpId.ValueString()
+	networkInterfaceId := model.NetworkInterfaceId.ValueString()
+	ctx = tflog.SetField(ctx, "project_id", projectId)
+	ctx = tflog.SetField(ctx, "public_ip_id", publicIpId)
+	ctx = tflog.SetField(ctx, "network_interface_id", networkInterfaceId)
+
+	core.LogAndAddWarning(ctx, &resp.Diagnostics, "stackit_public_ip_associate resource should never be used together with the stackit_public_ip resource.",
+		`Both resources have control of the network_interface association. If used together, this will lead to conflicts.`)
```

Comment:
```suggestion
	core.LogAndAddWarning(ctx, &resp.Diagnostics, "The stackit_public_ip_associate resource should never be used together with the stackit_public_ip resource.",
		`Both resources have control of the stackit_network_interface association. If used together, this will lead to conflicts.`)
```